software/Test_Robot_Software/main.py

```diff
@@ -0,0 +1,93 @@
+from camera import RealsenseCamera
+from image_processor import ImageProcessor
+from api import RobotMovement
+from robot_state import State
+from state_machine import StateMachine
+from time import time
+
+import cv2
+
+class Main:
+    """
+    This class combines all tasks:
+    finding a ball, following a ball, etc
+    """
+
+    def __init__(self, robot_movement):
+
+        self.debug = True
+        self.cam = RealsenseCamera(exposure=100)
+        self.processor = ImageProcessor(self.cam, debug=self.debug)
+        self.processor.start()        
+
+        self.robot_movement = RobotMovement()
+        self.state_machine = StateMachine(self.robot_movement)
+        self.current_state = State.FIND_A_BALL
```

Comment:
You should decide if the main or the state machine holds the current robot state.